src/Symfony/Component/Workflow/Registry.php

```diff
@@ -23,19 +24,34 @@ class Registry
 
     /**
      * @param Workflow $workflow
-     * @param string   $className
+     * @param string|SupportStrategyInterface $supportStrategy
      */
-    public function add(Workflow $workflow, $className)
+    public function add(Workflow $workflow, $supportStrategy)
     {
-        $this->workflows[] = array($workflow, $className);
+        if (!$supportStrategy instanceof SupportStrategyInterface) {
+            if (!is_string($supportStrategy)) {
+                throw new InvalidArgumentException('Expecting instance of SupportStrategyInterface or class name as a string.');
```

Comment:
If I am not mistaken, we don't do similar checks anywhere else, but rely on the fact that we claimed through the docblock which types we expect.